simplex.py

```diff
@@ -195,25 +197,20 @@ def run(self, r_data=None):
             ref_ff = self.ff.__class__()
             ref_ff.method = 'REFLECTION'
             ref_ff.params = copy.deepcopy(best_ff.params)
+            # Need score difference sum for weighted inversion. If zero, should break
+            score_diff_sum=sum([x.score - self.new_ffs[-1].score
+                                 for x in self.new_ffs[:-1]])
+            if score_diff_sum==0.:
+                logger.warning(
+                    'No difference between force field scores, breaking.')
+                break
             for i in xrange(0, len(best_ff.params)):
                 if self.do_weighted_reflection:
-                    try:
```

Comment:
I changed  it because your exception didn’t break the while, it broke instead the closest enclosing for-loop, and thus continued when the while should have been broken. I thought I wrote that in the commit message?

From: Eric Hansen [mailto:notifications@github.com]
Sent: den 28 juni 2016 16:19
To: ericchansen/q2mm <q2mm@noreply.github.com>
Cc: Norrby, Per-Ola <Per-Ola.Norrby@astrazeneca.com>; Author <author@noreply.github.com>
Subject: Re: [ericchansen/q2mm] Mysimplex (#36)


In simplex.py<https://github.com/ericchansen/q2mm/pull/36#discussion_r68765037>:

>              for i in xrange(0, len(best_ff.params)):

>                  if self.do_weighted_reflection:

> -                    try:

What you added in, the score_diff_sum is the same as what I have here.

Yours 1st checks whether that is zero, whereas mine simply breaks if it is zero. However, your method simply breaks without writing the best FF. This is bad and will break simplex from writing out the best current results. An error must be raised, such that opt.catch_run_errors can figure out how to handle the situation.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/ericchansen/q2mm/pull/36/files/0b9c5d0ad0a8d1105e4d26f192324c8bf323a928..398b4e1b8e1242703edfcc5b2a40b20047233942#r68765037>, or mute the thread<https://github.com/notifications/unsubscribe/ANVSNBGFQ4RGPzF2SpBGisIgn9jzLze5ks5qQS1KgaJpZM4JAC7r>.

________________________________

Confidentiality Notice: This message is private and may contain confidential and proprietary information. If you have received this message in error, please notify us and remove it from your system and note that you must not copy, distribute or take any action in reliance on it. Any unauthorized use or disclosure of the contents of this message is not permitted and may be unlawful.